test/test-TippyTopProvider.js

```diff
@@ -0,0 +1,93 @@
+"use strict";
+
+const {TippyTopProvider} = require("lib/TippyTopProvider");
+
+exports["test TippyTopProvider init"] = function*(assert) {
+  // Test the default init
+  let tippyTopProvider = new TippyTopProvider();
+  let site = {url: "http://somecraycrayurl.com"};
+  assert.deepEqual(site, tippyTopProvider.processSite(site));
+
+  // Test init with options
+  tippyTopProvider = new TippyTopProvider({
+    sites: [{
+      url: "https://mozilla.org",
+      image_url: "mozilla-org.png",
+      background_color: "#fff"
+    }]
+  });
+  assert.deepEqual(site, tippyTopProvider.processSite(site));
+};
+
+exports["test TippyTopProvider processSite"] = function*(assert) {
```

Comment:
can you add a test for a subdomain other than `www`? (even if it means no match for now)